Remove WordPressShared from CocoaPods by folding it

mokagio opened the pull request:
## Description
Similar to #15664 , but for WordPressShared.

The code came from
https://github.com/wordpress-mobile/WordPress-iOS-Shared/commit/6e8a44120452b08075d1838506bd632a5f83b044

## Steps to reproduce && Testing information
<!-- This is your opportunity to break out individual scenarios that need testing (when necessary) and/or include a checklist for the reviewer to go through. Consider documenting the following from your own completed testing: devices used, alternate workflows, edge cases, affected areas, critical flows, areas not tested, and any remaining unknowns. Provide feedback on this new section of the PR template through Sept 30, 2024 to Apps Quality; additional context here: p91TBi-b8z-p2#comment-12036 -->
See green CI and take the prototype build for a spin.

## Screenshots
<!-- Include before and after images or gifs when appropriate. -->

---
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.t

mokagio committed:
Remove WordPressShared from CocoaPods by folding the sources in the project

The code came from
https://github.com/wordpress-mobile/WordPress-iOS-Shared/commit/6e8a44120452b08075d1838506bd632a5f83b044

dangermattic commented:
<!--
  0 Errors
  3 Warnings: <code>Modules/Package.swift</c...
  0 Messages
  0 Markdowns
-->
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          3 Warnings
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><code>Modules/Package.swift</code> was changed without updating its corresponding <code>Package.resolved</code>. Please resolve the Swift packages as appropriate to your project setup (e.g. in Xcode or by running <code>swift package resolve</code>).</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">View files have been modified, but no screenshot or video is included in the pull request. Consider adding some for clarity.</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is larger than 300 lines of changes. Please consider splitting it into smaller PRs for easier and faster reviews.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

staskus reviewed (approved):
WordPressShared not really shared anymore. It'll be interesting to find out how much of it we really use.

mokagio committed:
Disable module verifier for WordPressAuthenticator

Otherwise, we get build errors on the WordPressSharedObjC import. Given
we don't ship this framework in isolation, we can deal with the lack of
verification as long as we know it builds in the app.
